huaweicloud/services/swr/resource_huaweicloud_swr_domain.go

```diff
@@ -0,0 +1,192 @@
+package swr
+
+import (
+	"context"
+	"time"
+
+	"github.com/chnsz/golangsdk/openstack/swr/v2/domains"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+)
+
+func ResourceSWRDomain() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceSWRDomainCreate,
+		ReadContext:   resourceSWRDomainRead,
+		UpdateContext: resourceSWRDomainUpdate,
+		DeleteContext: resourceSWRDomainDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(2 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"organization": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"repository": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"access_domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"permission": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"read",
+				}, false),
+			},
```

Comment:
If the permission is required and only support `read` as the value, maybe we can delete it from schema and coding it at `createOpts`

Reply:
Another point of view, we can make it optional, and set the default value to `read`, while removing the valid values check. This is more scalable. If more valid values are supported in the future, we can support them without modification.

Reply:
From this point of your view, I suggest not displaying this field or moving this field to the attribute parameter list in the document.

Reply:
From this point of your view, I suggest not displaying this field or moving this field to the attribute parameter list in the document.

Reply:
I think make it an optional parameter is better. In this way, we can get the balance between current convenience and 
future flexibility